workflow-controller/controller-service/src/main/java/Converters/Converter.java

```diff
@@ -0,0 +1,116 @@
+ * performs command line input validation and assigns to the respective field.
+ */
+public abstract class Converter implements CommandLine.ITypeConverter<List<File>> {
+
+    private static final Logger logger = LoggerFactory.getLogger(Converter.class);
+
+    public abstract String getFilePattern();
+
+    public abstract String getReferenceSchema();
+
+    public abstract String getWarnMessage();
+
+    public abstract int isDataValid(File inputFile) throws Exception;
+
+    public abstract Activity getDataMistmatchActivity();
+
+    /**
+     *  performs command line input validation and assigns to the respective field.
+     *
+     *  1.if input found empty throws Empty file path Hyscale Exception.
+     *  2.if input file does not exits throws File not found Hyscale Exception.
+     *  3.if input is not file throws Cannot process file Hyscale Exception.
+     *  4.if input does not satisfy the respective reference schema throws Invalid Format Hyscale Exception.
+     * @param inputFilePath
+     * @return assigns the file to field if valid.
+     * @throws Exception contains wrapped HyscaleExcpetion for Handling.
+     */
+    @Override
+    public List<File> convert(String inputFilePath) throws Exception {
+
+        List<File> inputFiles = new ArrayList<File>();
+
+        if (StringUtils.isBlank(inputFilePath)) {
+            WorkflowLogger.error(ControllerActivity.EMPTY_FILE_PATH);
+            throw new HyscaleException(CommonErrorCode.EMPTY_FILE_PATH);
+        }
+
+        File file = new File(inputFilePath);
+        if (!file.exists()) {
+            WorkflowLogger.error(ControllerActivity.CANNOT_FIND_FILE, inputFilePath);
+            throw new HyscaleException(CommonErrorCode.FILE_NOT_FOUND);
+        }
+        if (file.isFile()) {
+            String fileName = file.getName();
+            if (!fileName.matches(getFilePattern())) {
+                logger.warn(getWarnMessage(),inputFilePath.substring(inputFilePath.indexOf(ToolConstants.FILE_SEPARATOR) + 1));
+            }
+
+            if (isDataValid(file) == 1) {
+                logger.warn(getDataMistmatchActivity().getActivityMessage(), fileName);
+            }
+            ProcessingReport processingReport = SchemaProcessor.isSpecValid(file, getReferenceSchema(), ServiceInputType.YAML);
+            if (processingReport.isSuccess()) {
+                inputFiles.add(file);
+                return inputFiles;
+            }
+            Iterator<ProcessingMessage> messageIterator = processingReport.iterator();
+            while (messageIterator.hasNext()) {
+                ProcessingMessage message = messageIterator.next();
+                WorkflowLogger.error(ServiceSpecActivity.ERROR, message.getMessage());
+                logger.error(message.getMessage());
+                logger.error(String.valueOf(message.asException().fillInStackTrace()));
+                throw new HyscaleException(ServiceSpecErrorCodes.INVALID_FORMAT, message.getMessage());
```

Comment:
You are iterating over a loop and throwing an error inside loop indicating loop is always executed once. Please check the functional logic of the usecase.

Reply:
yes since i need to get the message from the process,iterator returns process message.